src/trio/_tests/test_unix_pipes.py

```diff
@@ -279,4 +279,12 @@ async def test_bizarro_OSError_from_receive() -> None:
 
 @skip_if_fbsd_pipes_broken
 async def test_pipe_fully() -> None:
-    await check_one_way_stream(make_pipe, make_clogged_pipe)
+    def make_pipe_async() -> tuple[FdStream, FdStream]:
+        await _core.checkpoint()
+        return make_pipe()
+
+    def make_clogged_pipe_async() -> tuple[FdStream, FdStream]:
+        await _core.checkpoint()
+        return make_clogged_pipe()
+
+    await check_one_way_stream(make_pipe_async, make_clogged_pipe_async)
```

Comment:
needing to duplicate `make_[clogged_]pipe` and having worthless checkpoints in them both looks awful. I vote for reverting

Reply:
Reverted in new version